### PR TITLE
Task-58297: Move RemoteSigningKeyResolver from mfa-oidc to gatein-portal

### PR DESCRIPTION
**Pull request by hchaabene**
Refactor OIDC protocol 
i moved  class RemoteSigningKeyResolver from mfa-oidc to gatein-portal

### CODE REVIEW

File: component/web/oauth-common/pom.xml [hunk 1]
```diff
@@ -137,8 +137,33 @@
...
+    </dependency>
+    <dependency>
+      <groupId>javax.json</groupId>
+      <artifactId>javax.json-api</artifactId>
+      <version>1.1</version>
+    </dependency>
+    <dependency>
+      <groupId>org.glassfish</groupId>
+      <artifactId>javax.json</artifactId>
+      <version>1.0.4</version>
```
Comment (rdenarie) [inline component/web/oauth-common/pom.xml]: you should not haev a version here

File: component/web/oauth-common/src/main/java/org/gatein/security/oauth/service/OIDCService.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
+package org.gatein.security.oauth.service;
```
Comment (rdenarie) [inline component/web/oauth-common/src/main/java/org/gatein/security/oauth/service/OIDCService.java]: Not needed class

File: component/web/oauth-common/src/main/java/org/gatein/security/oauth/service/MfaSystemService.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package org.gatein.security.oauth.service;
```
Comment (rdenarie) [inline component/web/oauth-common/src/main/java/org/gatein/security/oauth/service/MfaSystemService.java]: Not needed class

File: component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java [hunk 4]
```diff
@@ -106,6 +108,7 @@ public OpenIdProcessorImpl(ExoContainerContext context, InitParams params, Secur
         }
 
         this.secureRandomService = secureRandomService;
+        this.oidcService = oidcService;
```
Comment (rdenarie) [inline component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java]:
```suggestion

```

File: component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java [hunk 5]
```diff
@@ -21,6 +20,7 @@
 import org.gatein.security.oauth.common.OAuthConstants;
 import org.gatein.security.oauth.exception.OAuthException;
 import org.gatein.security.oauth.exception.OAuthExceptionCode;
+import org.gatein.security.oauth.service.OIDCService;
```
Comment (rdenarie) [inline component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java]:
```suggestion
```

File: component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java [hunk 6]
```diff
@@ -57,7 +57,9 @@ public class OpenIdProcessorImpl implements OpenIdProcessor {
 
     private final SecureRandomService secureRandomService;
 
-    public OpenIdProcessorImpl(ExoContainerContext context, InitParams params, SecureRandomService secureRandomService) {
+    private  final  OIDCService oidcService;
```
Comment (rdenarie) [inline component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java]:
```suggestion
```

File: component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java [hunk 7]
```diff
@@ -57,7 +57,9 @@ public class OpenIdProcessorImpl implements OpenIdProcessor {
...
+
+    public OpenIdProcessorImpl(ExoContainerContext context, InitParams params, SecureRandomService secureRandomService, OIDCService oidcService) {
```
Comment (rdenarie) [inline component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java]:
```suggestion
    public OpenIdProcessorImpl(ExoContainerContext context, InitParams params, SecureRandomService secureRandomService) {
```

File: component/web/oauth-common/pom.xml [hunk 8]
```diff
@@ -137,8 +137,33 @@
...
+    </dependency>
+    <dependency>
+      <groupId>io.jsonwebtoken</groupId>
+      <artifactId>jjwt-jackson</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>javax.json</groupId>
+      <artifactId>javax.json-api</artifactId>
+      <version>1.1</version>
```
Comment (rdenarie) [inline component/web/oauth-common/pom.xml]: no version here

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/RemoteJwkSigningKeyResolver.java [hunk 9]
```diff
@@ -0,0 +1,127 @@
+package org.gatein.security.oauth.openid;
```
Comment (rdenarie) [inline component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/RemoteJwkSigningKeyResolver.java:16]: Can you add the correct header for meeds ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — APPROVED** [review]

**Comment by hchaabene** [comment]
Refactor OIDC protocol
i moved class RemoteSigningKeyResolver from mfa-oidc to gatein-portal

**Pull request merged**